AX remote tokens are not being sent to iframe's WebContent process

pvollan opened the pull request:
#### af7cce33a111f6a364bcc5976f53123374a93af7
<pre>
AX remote tokens are not being sent to iframe&apos;s WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=252698">https://bugs.webkit.org/show_bug.cgi?id=252698</a>
rdar://problem/105750275

Reviewed by NOBODY (OOPS!).

Make sure the remote element and window AX tokens are being sent to the iframe&apos;s WebContent process.

* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::commitProvisionalFrame):
(WebKit::PageClientImpl::remoteWindowTokenForAccessibility):
(WebKit::PageClientImpl::remoteElementTokenForAccessibility):
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::remoteWindowTokenForAccessibility):
(WebKit::PageClientImpl::remoteElementTokenForAccessibility):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::remoteElementToken):
(WebKit::WebViewImpl::remoteWindowToken):
(WebKit::WebViewImpl::accessibilityRegisterUIProcessTokens):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af7cce33a111f6a364bcc5976f53123374a93af7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110855 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19941 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43493 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/2209 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119731 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21421 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11163 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/1994 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116607 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15908 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99106 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/103367 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97868 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30771 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44313 "") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12535 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32107 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/86192 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13092 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9082 "Found 1 new test failure: fast/images/avif-image-decoding.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18493 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51728 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15037 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/PageClient.h:
```diff
@@ -321,6 +321,11 @@ class PageClient : public CanMakeWeakPtr<PageClient> {
 #endif
 #endif
 
+#if PLATFORM(COCOA)
```
PLATFORM(MAC)?  There's just a stub on iOS.

pvollan replied:
Will fix!

Thanks for reviewing!

achristensen07 commented on Source/WebKit/UIProcess/WebFrameProxy.cpp:
```diff
@@ -408,6 +409,11 @@ void WebFrameProxy::commitProvisionalFrame(FrameIdentifier frameID, FrameInfoDat
         m_subframePage = makeUnique<SubframePageProxy>(*this, *m_page, m_process);
         m_page->didCommitLoadForFrame(frameID, WTFMove(frameInfo), WTFMove(request), navigationID, mimeType, frameHasCustomContentProvider, frameLoadType, certificateInfo, usedLegacyTLS, privateRelayed, containsPluginDocument, hasInsecureContent, mouseEventPolicy, userData);
+#if PLATFORM(COCOA)
+        auto elementToken = m_page->pageClient().remoteElementTokenForAccessibility();
```
Should this be in the ProvisionalFrameProxy constructor, or do we need the tokens to not be there until the load is committed?

pvollan replied:
> Should this be in the ProvisionalFrameProxy constructor, or do we need the tokens to not be there until the load is committed?

I believe we need to wait until the WebContent process has registered with Accessibility, which happens when the WebPage is created. I believe sending the tokens here will make sure that they are received after the registration.

achristensen07 reviewed (commented):
Is this testable?  What does this enable, or is it just progress in a good direction?

pvollan committed:
AX remote tokens are not being sent to iframe's WebContent process
https://bugs.webkit.org/show_bug.cgi?id=252698
rdar://problem/105750275

Reviewed by NOBODY (OOPS!).

Make sure the remote element and window AX tokens are being sent to the iframe's WebContent process.

* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::commitProvisionalFrame):
(WebKit::PageClientImpl::remoteWindowTokenForAccessibility):
(WebKit::PageClientImpl::remoteElementTokenForAccessibility):
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::remoteWindowTokenForAccessibility):
(WebKit::PageClientImpl::remoteElementTokenForAccessibility):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::remoteElementToken):
(WebKit::WebViewImpl::remoteWindowToken):
(WebKit::WebViewImpl::accessibilityRegisterUIProcessTokens):
